src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.PackageDependencyResolution.targets

```diff
@@ -172,12 +176,28 @@ Copyright (c) .NET Foundation. All rights reserved.
   <UsingTask TaskName="Microsoft.NET.Build.Tasks.ResolvePackageAssets"
             AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
 
+  <!-- Reads the version of the compiler APIs that are currently being used in order to pick the correct Roslyn components. -->
+  <Target Name="_ResolveCompilerVersion"
```

Comment:
Can we change this so that the code to read the compiler API version is done as part of the `ResolvePackageAssets` task instead of computed beforehand and passed in?  That way if we don't have any references to multi-targeted analyzers we don't even have to read the API version from the assembly, so we don't have to worry about the perf impact.

In that case probably you would need to pass in the path to the CodeAnalysis DLL to the task instead of the version, which would make testing harder.

Reply:
The idea we had offline was that the Roslyn targets would be setting the `$(CompilerApiVersion)` property statically. Thus this target wouldn't need to run at all.

cc @chsienki 

> That way if we don't have any references to multi-targeted analyzers

We are using this property to figure out if we have any references to multi-targeted analyzers at all. Using the first part of this property's value (`roslyn`), we are looking for folders that start with that name in the `analyzers` folder. I didn't want to hard-code "roslyn" into the C# task. It seems like a better design to put that information up in the .targets file, where it can be changed easily, if needed.